src/leap/mail/imap/service/imap.py

```diff
@@ -103,13 +181,8 @@ def buildProtocol(self, addr):
         # TODO should reject anything from addr != localhost,
         # just in case.
         log.msg("Building protocol for connection %s" % addr)
-        imapProtocol = self.protocol(
-            uuid=self._uuid,
-            userid=self._userid,
-            soledad=self._soledad)
-        imapProtocol.theAccount = self.theAccount
+        imapProtocol = self.protocol(self._soledad_sessions)
         imapProtocol.factory = self
```

Comment:
Do `LocalSoledadIMAPServer` need to know about the factory? I think is not used anywhere.